ft-datastore/src/main/java/org/ft/datastore/util/FeaturePropertyValidatorImpl.java

```diff
@@ -0,0 +1,56 @@
+package org.ft.datastore.util;
+
+import lombok.AllArgsConstructor;
+import lombok.NoArgsConstructor;
+import org.ft.core.api.model.FeatureInfo;
+import org.ft.core.util.FeaturePropertyValidator;
+import org.ft.datastore.models.Feature;
+import org.ft.datastore.models.FeatureStatus;
+import org.ft.datastore.repository.FeatureStatusRepository;
+import org.ft.datastore.repository.FeatureToggleRepository;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+
+@Service
+@AllArgsConstructor
+@NoArgsConstructor
+public class FeaturePropertyValidatorImpl implements FeaturePropertyValidator
+{
+    private FeatureStatusRepository featureStatusRepository;
+
+    private FeatureToggleRepository featureToggleRepository;
+
+    public boolean isValid (FeatureInfo featureInfo)
+    {
+        return featureInfo.validate();
+    }
+
+    @Override
+    public boolean validateDependencyForFeatureEnablement (String featureId,
+                                                           String tenant)
+    {
+
+        Optional<Feature> feature = featureToggleRepository.findById(featureId);
+        if (!feature.isPresent()) {
+            return false;
+        }
+
+        List<String> featureIdList = featureToggleRepository.getDependentFeatureIds(
```

Comment:
we already have the `Feature` instance we can directly get the`dependsOn` value by calling `feature.get().getDependsOn()`